REPRODUCTION_EXAMPLE.md

```diff
@@ -0,0 +1,151 @@
+# SharedResourceWarning Bug Reproduction Example
+
+## Issue #24477 - Working Example
+
+### Problem Description
+ArgoCD incorrectly triggers `SharedResourceWarning` for resources with the same name deployed in different clusters.
+
+### Reproduction Steps
+
+#### 1. Setup Multi-Cluster Environment
+
+**Cluster A Configuration:**
```

Comment:
How are you deploying Argo CD? You wouldn't deploy an argocd-cm to all clusters, just the one(s) where Argo CD is deployed.